lib/datadog/open_feature/transport/http/client.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative '../../../core/transport/http/env'
+require_relative '../../../core/transport/http/response'
+
+module Datadog
+  module OpenFeature
+    module Transport
+      module HTTP
+        # Routes, encodes, and sends tracer data to the trace agent via HTTP.
+        class Client
+          attr_reader :api, :logger
+
+          def initialize(api, logger: Datadog.logger)
+            @api = api
+            @logger = logger
+          end
+
+          def send_request(request)
+            env = build_env(request)
+
+            yield(api, env)
+          rescue => e
+            message = "Internal error during #{self.class.name} request. Cause: #{e.class.name} #{e.message} " \
```

Comment:
Do we need to include `self.class.name` here in the message, as we are already reporting the partial backtrace?